### PR TITLE
/read/subscriptions: Add 'Export' subscribers button

### LINKED ISSUE #78370: Subscription Management: Add ability to export subscriptions for logged-in users.
Add ability to export subscriptions for logged-in users.

It should follow the same behaviour and reuse the same API call that already existed from Reader.

Figma: inDLaEQV8jJ21O4WXawIsJ-fi-712_16894

**Subscription Management's Import/Export Dropdown:**
image

**Reader's Import/Export Dropdown:**
image

### PR DESCRIPTION
**Pull request by mashikag**
Closes https://github.com/Automattic/wp-calypso/issues/78370

## Proposed Changes

* Add `ReaderExportButton` to the header's ellipsis menu on the /read/subscriptions page
* Convert `ReaderExportButton` from .jsx class component to .tsx functional component
* Make`ReaderExportButton` component make use of `@wordpress/components` `Button` instead of the unmaintained `@automattic/components` `Button`

## Testing Instructions

1. Test Reader's new site subscriptions manager
   - Go to http://calypso.localhost:3000/read/subscriptions
   - Test the newly added export button 
2. Test Reader's old site subscriptions manager
   - Go to http://calypso.localhost:3000/following/manage
    Screenshot 2023-06-23 at 13 42 05
   - Ensure there are no regressions, especially with the "Export" feature
3. Test Reader's "Lists"
   - See the below GIF of how to get to the "Export" view in the Reader's "Lists". You may have to create your list first, if you do not have one yet.
    

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/reader/icons/download-cloud.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		</g>
+		<defs>
+			<clipPath id="clip0_3603_1982">
+				<rect width="20" height="20" fill="white" />
+			</clipPath>
+		</defs>
+	</svg>
+);
+
+export default downloadCloud;
```
Comment (yansern) [inline client/reader/icons/download-cloud.tsx:37]: Usually SVGs in script files need to be typed with things like `import { G, Path, SVG } from '@wordpress/components';`.

**Review by yansern — APPROVED** [review]
Tests well so I'm approving. I would suggest using the typed SVGs, not sure if you're gonna get lint errors.

(Also, I'm planning to insert tracks event for import/export too so we can track usage. Right now this is not being tracked.)

File: client/reader/icons/download-cloud.tsx (hunk 1 above)
Comment (mashikag) [reply client/reader/icons/download-cloud.tsx:37]: Oh, interesting. I was not aware of those. Thanks! 🙇

**Comment by mashikag** [comment]
> I would suggest using the typed SVGs

I will push a commit now to use the components. Though, I am not sure what is the benefit of using those. 🤔 

> not sure if you're gonna get lint errors.

FYI, there is no such lint rules defined. 👌

**Pull request merged**